src/main/java/ch/njol/skript/expressions/ExprFilter.java

```diff
@@ -61,6 +61,20 @@ public class ExprFilter extends SimpleExpression<Object> {
 	static {
 		Skript.registerExpression(ExprFilter.class, Object.class, ExpressionType.COMBINED,
 				"%objects% (where|that match) \\[<.+>\\]");
+
+		ExprLoopValue.registerLoopValueHandler(ExprFilter.class, (source, type) -> {
+			for (ExprInput<?> child : source.children) { // if they used player input, let's assume loop-player is valid
+				//noinspection ConstantConditions
+				if (child.getClassInfo() == null || child.getClassInfo().getUserInputPatterns() == null)
+					continue;
+
+				for (Pattern pattern : child.getClassInfo().getUserInputPatterns()) {
+					if (pattern.matcher(type).matches())
+						return true;
+				}
+			}
+			return ExprLoopValue.isLoopOf(source.objects, type); // Nothing matched, so we'll rely on the object expression's logic
+		});
```

Comment:
This feels like weird behaviour for this expression, but that may be due to the ExprInput accepting all types without question:
![image](https://user-images.githubusercontent.com/29547183/187085092-47aeb32d-33bd-4c5e-9890-275263199032.png)
parses just fine

Reply:
However, if you were to change that behaviour to only allow using `player input` in the above example, wouldn't this loop be useless (only allowing player, which was already allowed by the default impl)

Reply:
I can't speak as to why it tries to rely on the expression as a backup - but I do agree that is odd behavior we shouldn't allow. Seems like something we should try to fix with ExprInput, no?

Reply:
Yea we should fix that in ExprInput

Reply:
I think the rationale was basically "just in case" when I added that... so basically no real reason xD

Reply:
From what I can tell there is no reason for a loop handler to exist here. It should only be based on the actual type being looped, which as TP mentioned, is covered by default. Another fix to not accept any value without care should be made for ExprFilter